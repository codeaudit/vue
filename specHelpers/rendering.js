import toJson from 'enzyme-to-json';
import React, {PropTypes} from 'react';
import {combineReducers, createStore} from 'redux';
import {mount, render, shallow} from 'enzyme';
import {reducer as uiReducer} from 'redux-ui';
import theme from './TestTheme';

const reducer = combineReducers({
    ui: uiReducer,
});
const store = createStore(reducer);

export const getShallow = (Component) => (props = {}, context = {
    store,
    theme,
}) => shallow(<Component {...props} />, {context});

export const getRender = (Component) => (props = {}, context = {
    store,
    theme,
}) => render(<Component {...props} />, {
    context,
    childContextTypes: {
        store: PropTypes.object,
        theme: PropTypes.object,
    },
});

export const getMount = (Component) => (props = {}, context = {
    store,
    theme,
},) => mount(<Component {...props} />, {
    context,
    childContextTypes: {
        theme: PropTypes.object,
    },
});

export const snapshot = (renderedComponent) => toJson(renderedComponent);
