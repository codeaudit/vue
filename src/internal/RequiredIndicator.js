import React, {Component, PropTypes} from 'react';
import * as Opacity from './../utilities/Opacity';
import Radium from './../utilities/Radium';

class RequiredIndicator extends Component {
    static propTypes = {hidden: PropTypes.bool};

    static defaultProps = {hidden: false};

    static contextTypes = {
        theme: PropTypes.shape({
            color: PropTypes.shape({requiredPrimary: PropTypes.string}),
            typography: PropTypes.shape({
                lineHeightNormal: PropTypes.number,
                small: PropTypes.number
            })
        })
    };

    constructor(...args) {
        super(...args);
        this.getStyles = this.getStyles.bind(this);
    }

    render() {
        const styles = this.getStyles();
        return (
            <div style={styles.root}>*</div>
        );
    }

    getStyles() {
        const {
            color: {requiredPrimary},
            typography: {
                lineHeightNormal,
                small
            }
        } = this.context.theme;
        const {hidden} = this.props;
        const zIndex = 1;

        return {
            root: {
                alignSelf: 'center',
                color: requiredPrimary,
                fontSize: small,
                lineHeight: lineHeightNormal,
                opacity: hidden ? Opacity.hidden : Opacity.fullyVisible,
                zIndex
            }
        };
    }
}
export default Radium(RequiredIndicator);
