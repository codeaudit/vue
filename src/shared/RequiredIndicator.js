import React, {Component, PropTypes} from 'react';
import * as CustomPropTypes from '../utilities/PropTypes';

const getStyles = (theme, props) => {
    // TODO: pull out into theme/css/etc.
    const color = 'red';

    return {
        root: {
            color,
            ...props.style,
            opacity: props.hidden ? 0 : 1,
            zIndex: 1,
            transition: 'all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms'
        }
    };
};

class RequiredIndicator extends Component {
    static propTypes = {
        hidden: PropTypes.bool,
        style: CustomPropTypes.style
    };

    static defaultProps = {
        hidden: false,
        style: {}
    };

    render() {
        const styles = getStyles(this.context.theme, this.props);
        return (
            <div style={styles.root}>*</div>
        );
    }
}
export default RequiredIndicator;
