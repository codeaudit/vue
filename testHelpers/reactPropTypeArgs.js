let ReactPropTypesSecretExtraArgs;
try {
    ReactPropTypesSecretExtraArgs = [
        'location',
        'propFullName',
        require('react/lib/ReactPropTypesSecret'),
    ];
} catch (e) {
    ReactPropTypesSecretExtraArgs = [];
}

export default ReactPropTypesSecretExtraArgs;
