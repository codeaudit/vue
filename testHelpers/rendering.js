import toJson from 'enzyme-to-json';
import React, {PropTypes} from 'react';
import {mount, render, shallow} from 'enzyme';
import theme from './../packages/themes/src/V1Theme';

const ctx = {
    theme,
};

export const getShallow = (Component) => (props = {}, context = {}) => shallow(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
});

export const getRender = (Component) => (props = {}, context = {}) => render(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
});

export const getMount = (Component) => (props = {}, options = {}, context = {}) => mount(<Component {...props} />, {
    childContextTypes: {
        theme: PropTypes.object,
    },
    context: {
        ...ctx,
        ...context,
    },
    ...options,
});

const replaceUIKey = (componentName) => {
    let index = 0;
    return (json) => {
        if (!json.props || !json.props.uiKey) {
            return json;
        }
        const uiKey = `${componentName}-${index}`;
        index += 1;
        return Object.assign({}, json, {
            uiKey,
            uiPath: [uiKey],
        });
    };
};
const replaceUiKeyRecursively = (json) => {
    if (!json.type) {
        return json;
    }
    const replaceUIKeyInParent = replaceUIKey(json.type);
    const output = replaceUIKeyInParent(json);
    if (!output.children) {
        return json;
    }
    output.children = json.children.map((child, index) => replaceUiKeyRecursively(child));
    return output;
};

export const snapshot = (renderedComponent) => {
    return toJson(renderedComponent);
};

export const reset = (component) => () => {
    if (component) {
        component.unmount();
    }
    document.body.innerHTML = '';
};
