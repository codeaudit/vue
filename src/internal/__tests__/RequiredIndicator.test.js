import RequiredIndicator from '../RequiredIndicator';
import {getShallow, snapshot} from './../../../testHelpers/rendering';

const shallowRenderIndicator = getShallow(RequiredIndicator);

test('required indicator renders with defaults', () => {
    const component = shallowRenderIndicator();
    expect(snapshot(component)).toMatchSnapshot();
});

test('required indicator renders with defaults', () => {
    const component = shallowRenderIndicator({
        hidden: true,
    });
    expect(snapshot(component)).toMatchSnapshot();
});
