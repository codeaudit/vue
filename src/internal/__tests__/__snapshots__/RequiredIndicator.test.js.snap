exports[`test required indicator renders with defaults 1`] = `
<div
  data-radium={true}
  style={
    Object {
      "alignSelf": "center",
      "color": "#ffc0cb",
      "fontSize": "14px",
      "lineHeight": 1.285,
      "opacity": 1,
      "zIndex": 1,
    }
  }>
  *
</div>
`;

exports[`test required indicator renders with defaults 2`] = `
<div
  data-radium={true}
  style={
    Object {
      "alignSelf": "center",
      "color": "#ffc0cb",
      "fontSize": "14px",
      "lineHeight": 1.285,
      "opacity": 0,
      "zIndex": 1,
    }
  }>
  *
</div>
`;
